### ISSUE TITLE
Training with 1 source image possible?

### ISSUE THREAD

**Issue by DataSupply** [ultralytics/yolov5#10853]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to use yolo for recognizing emojis in images. So i've made a small test project for using a custom training set. However i always get "no detection" so it seems i am missing something..?

Here's what i did:

- Converted the default Apple emoji font to separate images (160x160 JPEG) for test used 5 images.
- Generated the labels with the structure contents CLASSNUMBER 0.5 0.5 1 1
- Created a yaml file:

`path: /LOCALPATH/datasets/emoji  # dataset
train: images  # train images (relative to 'path') 5 images
val: images  # val images (relative to 'path') 5 images
test:  # test images (optional)

# Classes
names:
        0: U00002764
        1: U0001F437
        2: U0001F920
        3: U000026C4
        4: U0001F996  `

- Trained using:

`python3 train.py --img 160 --batch 24 --epochs 100 --data data/emoji.yaml --weights models/yolov5s.pt`

- Detect test by running the detect script using an image that has the exact same emoji's pasted in:
`python3` detect.py --weights /LOCALPATH/yolov5/runs/train/exp20/weights/best.pt --source /LOCALPATH/test_emoji.jpg `

![U0001F920](https://user-images.githubusercontent.com/867099/215326536-45a7b258-81e0-44a9-9d17-1143a0269699.jpg)

![test_emoji](https://user-images.githubusercontent.com/867099/215326532-2818f1d1-b6aa-41ca-94c0-25ebb3342c6a.jpg)

These image are in the "Train/exp" folder which seems to me that the training went ok: 

Train_batch0.jpg:
![train_batch0](https://user-images.githubusercontent.com/867099/215326627-fb0de248-171a-40c7-9435-72bd7aa50bff.jpg)

val_batch0_labels.jpg:
![val_batch0_pred](https://user-images.githubusercontent.com/867099/215326634-07d07ef8-0b3f-4265-8097-115a21abe1a7.jpg)

val_batch0_pred.jpg:
![val_batch0_labels](https://user-images.githubusercontent.com/867099/215326638-b05bd29b-eff2-4fee-aed8-cf05880cb3c8.jpg)

results.png:
![results](https://user-images.githubusercontent.com/867099/215326641-346fa45d-61aa-48f5-bf3e-a3d45aed6753.png)

Hoping someone can point me in the direction solving this. Much appreciated!

**Comment by halskov** [comment]
Why do you want to train with only one image? Normally it is suggested to have at least a thousand images in your training set :)

The images you see in the train/exp folder might be the ground truth images and not actual detections. You will need to provide us the filenames for the images you present.

**Comment by DataSupply** [comment]
Thank you for the reply! I updated the original question with the file names. 

I do see why a 1000 training images for a class like "people" will work due to all the variants, however as the emoji's are really 'static' and don't have a lot of differences, i thought one source image should be enough. I can only think of some potential blur variants but other than that it don't understand why this simple example (with the test image including the original train images) does not already work?

**Comment by glenn-jocher** [comment]
@DataSupply i appreciate your approach, and it's understandable that you expected more success with a simpler dataset. Nonetheless, training on just one source image might not capture the necessary variations for accurate detection, even for static emojis. Incorporating more diverse instances could help your model recognize the emojis across various contexts. Additionally, considering some variations in the dataset, such as different sizes or rotations, might also improve the results.

The typical recommendation for a larger training set stems from the need to generalize the model for better real-world performance across different scenarios. If you'd like to discuss further options or need assistance with augmentation techniques, feel free to ask.
